pycbc/waveform/waveform.py

```diff
@@ -110,6 +110,18 @@ def _check_lal_pars(p):
         lalsimulation.SimInspiralWaveformParamsInsertFrameAxis(lal_pars, p['frame_axis'])
     if p['side_bands']:
         lalsimulation.SimInspiralWaveformParamsInsertSideband(lal_pars, p['side_bands'])
+    if p['mode_array']:
+        try:
+            ma = lalsimulation.SimInspiralCreateModeArray()
+            if p['mode_array']==-1:
+                lalsimulation.SimInspiralModeArrayActivateAllModes(ma)
+            else:
+                for l,m in p['mode_array']:
+                    lalsimulation.SimInspiralModeArrayActivateMode(ma, l, m)
+            lalsimulation.SimInspiralWaveformParamsInsertModeArray(lal_pars, ma)
+        except AttributeError:
```

Comment:
Why is the code coming here if mode_array is not provided? The error should only be raised if *using* the mode array .... I think this is pretty close now, just don't print an error if the user isn't giving a mode_array. (It isn't an error to use an earlier version of lalsuite .... Also don't *print* "ERROR" and then carry on, that feels really wrong to me. *Raise* the error, but only if the mode_array is given.)